projects/cdk/schematics/ng-update/v4/steps/migrate-proprietary.ts

```diff
@@ -0,0 +1,150 @@
+
+            sourceFile.replaceWithText(fullText);
+        });
+
+    fileSystem.commitEdits();
+    saveActiveProject();
+}
+
+function replaceAngularJsonStyles(options: TuiSchema, tree: Tree): Rule {
+    let proprietaryIcons: string | null = null;
+
+    try {
+        proprietaryIcons =
+            getPackageJsonDependency(tree, '@taiga-ui/proprietary-icons')?.version ||
+            getPackageJsonDependency(tree, '@taiga-ui/proprietary-tds-icons')?.version ||
+            null;
+
+        if (getSourceFiles('**/angular.json').length) {
+            return updateWorkspace(workspace => {
+                if (proprietaryIcons === null) {
+                    return;
+                }
+
+                const projects = getProjects(options, workspace);
+
+                if (!projects.length) {
+                    return;
+                }
+
+                projects.forEach(project => {
+                    let targetOptions;
+
+                    try {
+                        targetOptions = getProjectTargetOptions(project, 'build');
+                    } catch {
+                        return;
+                    }
+
+                    if (Array.isArray(targetOptions?.assets)) {
+                        targetOptions.assets = (targetOptions.assets as Asset[]).filter(
+                            asset => {
+                                const input = tuiIsString(asset) ? asset : asset.input;
+
+                                return (
+                                    !input.includes('@taiga-ui/proprietary-icons') &&
+                                    !input.includes('@taiga-ui/proprietary-tds-icons')
+                                );
+                            },
+                        );
+
+                        targetOptions.assets.push({
+                            glob: '**/*',
+                            input: '@taiga-ui/proprietary/design-tokens/icons/src',
+                            output: 'assets/taiga-ui/icons',
+                        });
+                    }
+                });
+            });
+        }
+    } catch {
+        // noop
+    }
+
+    return () => {};
+}
+
+export function removeProprietaryPackages() {
+    return (tree: Tree, context: SchematicContext): void => {
+        try {
+            const fileSystem = getFileSystem(tree);
+
+            [
+                '@taiga-ui/proprietary-core',
+                '@taiga-ui/proprietary-navigation',
+                '@taiga-ui/proprietary-banking',
+                '@taiga-ui/proprietary-icons',
+                '@taiga-ui/proprietary-tds-icons',
+                '@taiga-ui/proprietary-tds-palette',
+            ].forEach(name => removePackageJsonDependency(tree, name));
+
+            addPackageJsonDependency(tree, {
+                name: '@taiga-ui/proprietary',
+                version: TUI_VERSION,
+            });
+
+            context.addTask(new NodePackageInstallTask());
```

Comment:
Let's add this task `NodePackageInstallTask` in the bottom of main. We will need to remove the `experimental` package, so this task will always be useful, not only in proprietary projects

Reply:
fixed